postgres/toddl.go

```diff
@@ -216,3 +220,30 @@ func cvtForeignKeys(conv *internal.Conv, srcTable string, srcKeys []schema.Forei
 	}
 	return spKeys
 }
+
+func cvtIndexes(conv *internal.Conv, srcTable string, srcIndexes []schema.Index, schemaIndexKeys map[string]bool) []ddl.CreateIndex {
+	var spIndexes []ddl.CreateIndex
+	for _, srcIndex := range srcIndexes {
+		var spKeys []ddl.IndexKey
+		for _, k := range srcIndex.Keys {
+			spCol, err := internal.GetSpannerCol(conv, srcTable, k.Column, true)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Can't map index for table %s", srcTable))
+				continue
+			}
+			spKeys = append(spKeys, ddl.IndexKey{Col: spCol, Desc: k.Desc})
+		}
+		if srcIndex.Name == "" {
+			b := make([]byte, 4)
+			_, err := rand.Read(b)
```

Comment:
its an inbuilt function and ideally it won't throw any error, but as it is returning an error object, we are checking for possibility.

Reply:
SGTM, let's change to `if _, err := ...; err != nil` style here though. And add a comment after `if srcIndex.Name == ""` as following:

// Generate a random name if index name is empty in postgres since spanner requires globally unique non-empty index names.

Reply:
added comment and updated err statement

Reply:
not relevant anymore